steps/solvers.py

```diff
@@ -1,14 +1,17 @@
-import yaml
+import json
 from zquantum.core.utils import create_object
-from zquantum.qubo import load_qubo
+from zquantum.qubo import load_qubo, save_sampleset
 from zquantum.core.measurement import Measurements
 
 
-def solve_qubo(qubo, solver_specs, sample_params):
+def solve_qubo(qubo, solver_specs, sampler_params=None):
     """Solves qubo using any sampler implementing either dimod.Sampler or zquantum.qubo.BQMSolver"""
+    if sampler_params is None:
+        sampler_params = {}
     solver = create_object(solver_specs)
     qubo = load_qubo(qubo)
-    sampleset = solver.sample(qubo, **sample_params)
+    sampleset = solver.sample(qubo, **sampler_params)
     best_sample_dict = sampleset.first.sample
     solution_bitstring = tuple(best_sample_dict[i] for i in sorted(best_sample_dict))
     Measurements([solution_bitstring]).save("solution.json")
```

Comment:
could we append the `solver_specs["FunctionName"]` to the filenames? 
In a workflow that does both annealing and exact solvers, won't this give duplicate artifacts?

Reply:
No – you don't have get duplicate artifacts, because each step is separate.
So even though your files are called the same (`solution.json`), they are both in two separate parts of the output json corresponding to two different steps, corresponding to two different solvers.

Reply:
okay cool!